src/mscorlib/src/System/Collections/Generic/List.cs

```diff
@@ -1051,6 +1054,29 @@ public void Sort(IComparer<T> comparer)
             return true;
         }
 
+        private class PooledIEnumerator : IEnumerator<T>
+        {
+            private Enumerator _enumerator;
+
+            internal PooledIEnumerator Initalize(List<T> list)
+            {
+                _enumerator = new Enumerator(list);
+                return this;
+            }
+            public bool MoveNext() => _enumerator.MoveNext();
+            public T Current => _enumerator.Current;
+            object IEnumerator.Current => Current;
+            public void Dispose()
+            {
+                _enumerator = default(Enumerator);
```

Comment:
Is that common? You'd have to go via the interface rather than the concrete type and manually be calling Dispose. It would be hard to do the pooling otherwise without changing the compiler and doing escape analysis.